test/EzMap.IntegrationTest/AuthenticationControllerTest.cs

```diff
@@ -0,0 +1,119 @@
+
+public class AuthenticationControllerTest
+{
+    [Fact]
+    public async Task SignIn_CorrectUsernamePassword_TokenShouldBeGranted()
+    {
+        // arrange 
+        var app = new TestWebAppFactory<Program>();     
+        var client = app.CreateClient();
+        using var scope = app.Services.CreateScope();
+
+        var dbContext = scope.ServiceProvider.GetRequiredService<EzMapContext>();
+        dbContext.Add(new User("bytetum", "redbull", "bytetum@gmail.com",
+            BCrypt.Net.BCrypt.HashPassword("ThHo!0322")));
+        await dbContext.SaveChangesAsync();
+        // act
+        var response = await client.PostAsJsonAsync("api/user/signin",
+            new UserSignInDto()
+                { Username = "redbull", Password = "ThHo!0322" });
+
+        var content = await response.Content.ReadAsStringAsync();
+        // assert 
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        Assert.NotEmpty(content);
+    }
+
+    [Fact]
+    public async Task SignIn_ReturnsUnauthorizedError_WhenIncorrectCredentialProvided()
+    {
+        // arrange 
+        var app = new TestWebAppFactory<Program>();
+        var client = app.CreateClient();
+        using var scope = app.Services.CreateScope();
+
+        var dbContext = scope.ServiceProvider.GetRequiredService<EzMapContext>();
+        dbContext.Add(new User("bytetum", "redbull", "bytetum@gmail.com",
+            BCrypt.Net.BCrypt.HashPassword("ThHo!0322")));
+        await dbContext.SaveChangesAsync();
+        // act
+        var response = await client.PostAsJsonAsync("api/user/signin",
+            new UserSignInDto()
+                { Username = "redbull", Password = "ThHoThHo!" });
+
+        // assert 
+        Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+    }
+
+    [Fact]
+    public async Task SignUp_ProvideCorrectData_UserRecordCreated()
+    {
+        // arrange 
+        var app = new TestWebAppFactory<Program>();
+        var client = app.CreateClient();
+        using var scope = app.Services.CreateScope();
+        var user = new UserCreationDto()
+        {
+            UserName = "bytetum", DisplayName = "hoangml", Email = "notdeciced@gmail.com", Password = "ThHo|0322",
+        };
+
+        // act
+        // client send http request including a valid user 
+        var response = await client.PostAsJsonAsync("api/user/signup", user);
+
+
+        // assert
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+        var dbContext = scope.ServiceProvider.GetRequiredService<EzMapContext>();
+        var dbUser = dbContext.Users.FirstOrDefault(x =>
+            x.UserName == user.UserName && x.Email == user.Email && x.DisplayName == user.DisplayName);
+
+        // successful message returned
+        Assert.True(BCrypt.Net.BCrypt.Verify(user.Password, dbUser.Password));
+    }
+    
+    [Fact]
+    public async Task SignUp_InvalidPasswordViolatedThreeComplixityCheck_BadRequestAndThreeErrorMessagesReturn()
```

Comment:
This name is not sane. So if you add a check you'll have to rename the method to ...FourCheck...? I don't think future you or another developer will remember to do that. Either way, just make it a generic name without THE NUMBER please.